src/html_support_files/odoc.css

```diff
@@ -754,11 +781,174 @@ td.def-doc *:first-child {
   display: block;
 }
 
-.odoc-toc ul li a:hover {
+.odoc-sidebar ul li a:hover {
   box-shadow: none;
   text-decoration: underline;
 }
 
+.odoc-search {
+  position: sticky;
+  top: 0;
+  height: min-content;
+  background: var(--main-background);
+  width: 100%;
+  padding-top: 1rem;
+}
+
+
+.odoc-search .search-inner {
+  width: 100%;
+  position: relative;
+  left:0;
+  transition : left 0.3s, transform 0.3s, width 0.3s; 
+}
+
+.odoc-search:focus-within .search-inner {
+  left: 50%;
+  transform: translateX(-50%);
+  width: 110%;
+}
+
+.odoc-search .search-bar {
+  position: relative;
+  z-index: 2;
+  font-size: 1;
+  transition: font-size 0.3s;
+}
+
+.odoc-search:focus-within .search-bar {
+  font-size: 1.1em;
+}
+
+.odoc-search:not(:focus-within) .search-result {
+  display: none;
+}
+
+.odoc-search .search-result-inner:empty {
+  display: none;
+}
```

Comment:
When no results are found, it should be indicated to the user. This may be done here in css or in js.

Reply:
If possible also, it might be useful to render differently when search is in progress.

Reply:
Both are implemented in e72afb8